tests/EndToEndTests.hs

```diff
@@ -483,3 +509,106 @@ tests = testSpec "End-to-end tests" $ do
                 ]
               ]
         toJSON (toValue response) `shouldBe` expected
+
+  describe "introspection" $ do
+    treatRef <- runIO $ newIORef 1
+
+    let 
+      run :: Text -> IO Response
+      run query = interpretRequest @E2ESchema (schemaHandler mortgage treatRef) query Nothing mempty
+
+    it "can issue direct queries" $ do
+      response <- run [r|query myQuery {
+         dog {
+           name
+         }
+        }
+      |]
+      response `shouldBeJSON` [json|
+        {
+          "data": {
+            "dog": {
+              "name": "Mortgage"
+            }
+          }
+        }
+      |]
+
+    it "can issue mutations" $ do
+      response <- run [r|mutation goodBoy {
+           giveTreats(count: 2) {
+             dog {
+               name
+             }
+             treats
+           }
+         }
+      |]
+      response `shouldBeJSON` [json|
+        {
+          "data": {
+            "giveTreats": {
+              "dog": {
+                "name": "Mortgage"
+              },
+              "treats": 3
+            }
+          }
+        }
+      |]
+
+    -- it "can fetch the __schema" $ do
```

Comment:
Are the commented out tests aspirational?

Reply:
Yeah ... I'll fix / expand / uncomment (or remove) any of them before finalizing this for sure.